### PR TITLE
[Core]JdbcCatalog supports custom configuration for the length of lock table

### PR DESCRIPTION
**Pull request by sunxiaojian**
Linked issue: close (https://github.com/apache/incubator-paimon/issues/3046)

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Modify too more... I don't think we need to let all length configurable...

I think we can just modify this for lock...

**Comment by sunxiaojian** [comment]
> Modify too more... I don't think we need to let all length configurable...
> 
> I think we can just modify this for lock...

Okay,  but this way, when other fields exceed the limit, they can only be resolved through the alter table

**Comment by sunxiaojian** [comment]
> > Modify too more... I don't think we need to let all length configurable...
> > I think we can just modify this for lock...
> 
> Ok, but this way, when other fields exceed the limit, they can only be resolved through the alter table

File: docs/content/how-to/creating-catalogs.md [hunk 1]
```diff
@@ -210,8 +210,11 @@ You can configure any connection parameters that have been declared by JDBC thro
 
 You can also perform logical isolation for databases under multiple catalogs by specifying "catalog-key".
 
+Additionally, when creating a JdbcCatalog, you can specify the maximum length for the lock key by configuring "lock-key-max-length," which defaults to 255. Since this value is a combination of {catalog-key}.{database-name}.{table-name}, please adjust accordingly.
```
Comment (JingsongLi) [inline docs/content/how-to/creating-catalogs.md:213]: which defaults to 1000

File: docs/layouts/shortcodes/generated/jdbc_catalog_configuration.html [hunk 2]
```diff
@@ -32,5 +32,11 @@
             <td>String</td>
             <td>Custom jdbc catalog store key.</td>
         </tr>
+        <tr>
+            <td><h5>lock-key-max-length</h5></td>
+            <td style="word-wrap: break-word;">1000</td>
```
Comment (JingsongLi) [inline docs/layouts/shortcodes/generated/jdbc_catalog_configuration.html:37]: Here is 1000, but others are 255?

Comment (sunxiaojian) [reply docs/layouts/shortcodes/generated/jdbc_catalog_configuration.html]: > Here is 1000, but others are 255?

@JingsongLi fixed

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**